apps/opik-backend/src/main/java/com/comet/opik/domain/ChatCompletionService.java

```diff
@@ -43,10 +46,20 @@ public ChatCompletionResponse create(@NonNull ChatCompletionRequest request, @No
 
         ChatCompletionResponse chatCompletionResponse;
         try {
+            log.info("Creating chat completions, workspaceId '{}', model '{}'", workspaceId, request.model());
             chatCompletionResponse = retryPolicy.withRetry(() -> llmProviderClient.generate(request, workspaceId));
+            log.info("Created chat completions, workspaceId '{}', model '{}'", workspaceId, request.model());
         } catch (RuntimeException runtimeException) {
             log.error(UNEXPECTED_ERROR_CALLING_LLM_PROVIDER, runtimeException);
-            throw llmProviderClient.mapRuntimeException(runtimeException);
+            if (llmProviderClient.getHttpExceptionClass().isInstance(runtimeException.getCause())) {
+                int statusCode = llmProviderClient.getHttpErrorStatusCode(runtimeException);
+                if (statusCode >= 400 && statusCode <= 499) {
```

Comment:
you can use the following code to check for response families instead of doing this yourself:
Response.Status.Family.familyOf(statusCode) == Response.Status.Family.CLIENT_ERROR)

works well for success response as well (I think I saw the same type of comparison for 200 somewhere in your code)

Reply:
Done in https://github.com/comet-ml/opik/pull/957/commits/be10cfaf26e847f454216541138197915820b875